sktime/utils/ts_format.py

```diff
@@ -0,0 +1,189 @@
+# -*- coding: utf-8 -*-
+import pandas as pd
+import re
+
+__all__ = ["csv_to_ts_df"]
+
+__author__ = ["Christopher Holder"]
+
+
+def csv_to_ts_df(csv_df, group_by):
```

Comment:
I can't follow the names here, I understand csv, but I'm not sure about what format `ts_df` refers too. 

If it's the nested `pd.DataFrame`, I'd prefer calling it `nested` (i.e. `csv_to_nested`) as this is what we're using elsewhere.